### ISSUE TITLE
new espeak variants

### ISSUE THREAD

**Issue by joshknnd1982** [espeak-ng/espeak-ng#1649]
Attached is a file with many new espeak variants. Could you please include them in the next espeak version? thanks.
[exclamation-v.zip](https://github.com/espeak-ng/espeak-ng/files/10700071/exclamation-v.zip)

**Comment by joshknnd1982** [comment]
I noticed adding all the new variants in NVDA version 2022.4 causes several languages such as Italian to disappear although their files are still there in espeak ng. I recommend using a portable NVDA for now to try out the variants if you need languages such as italian.

**Comment by gregodejesus** [comment]
the same thing happens to me when I try to add more than 230 variants, by the way here I leave another one. felipe variant

**Comment by joshknnd1982** [comment]
I had the same issue so I went through the new variants file and deleted duplicate variants. I am now only adding 2 or 3 at a time. so far all languages are staying in place. the new cleaned up variants file is attached. this just has new variants, it does not 
[espeaknewvariants.zip](https://github.com/espeak-ng/espeak-ng/files/10709277/espeaknewvariants.zip)
have the already pre-existing variants.

**Comment by joshknnd1982** [comment]
maybe since we have many new klatt6 variants now and other klatt variants, it would be good for organizational purposes to separate variants into their own folders. By that I mean, klatt based variants go into a klatt variants folder and traditional non-klatt-variants do not. just a thought.

**Comment by joshknnd1982** [comment]
I was told to make a pull request. I tried and the developers closed it. sorry. unless the developers like Alexanders decide to put the new variants into espeak ng it looks like we are out of luck. Just enjoy using them with NVDA.

**Comment by hozosch** [comment]
Well there is a reason why it was closed, you didn't hand it in the proper way. Look at the thread for more details.

**Comment by joshknnd1982** [comment]
I tried making a pull request. I had to do some strange comparison thing between branches or forks as they call it. then it said i had to make my own copy of espeak ng, put the variant files into my copy and somehow in some way I don't understand I somehow have to merge my espeak ng fork into the main master repository. It's too confusing!! In the 1990s if you wanted to share code with someone you emailed them a zip file. It was a lot simpler back then. so guys enjoy the variants because unfortunately and I am sorry to say this and I don't mean to disappoint you, I don't think the devs on here really care. so if you want to use the new variants, just put them into C:\Program Files (x86)\NVDA\synthDrivers\espeak-ng-data\voices\!v    and they will work with NVDA. If you want to use them on android or iPhone, sorry you are out of luck because pull requests go way above my limited development knowledge of some basic jaws scripting commands. so enjoy what we have and accept the limitations of the espeak ng devs on here.

**Comment by joshknnd1982** [comment]
iff the variants get included, great. if not. fine. I give up. In the 1990s you gave people new code and variants in small zip files. I guess I'll just use them with NVDA and be done with it. My knowledge is limited. so I want to use the new variants, put them in the propper folder, restart NVDA and they work. If you want to use them on android or iPhone espeak then sorry you're out of luck I have no clue how to add variants to those espeak versions. It takes coding skill that is way beyond what i am able to do.

**Comment by 0x5066** [comment]
To quote myself:
That is not how a pull request works, you're supposed to fork the repository, upload all the files to their respective directory (which is /espeak-ng-data/voices/!v), then make the proper pull request. Attaching a zip file here is pointless because you are supposed to add these to the repository, not them.

**Comment by joshknnd1982** [comment]
when I use espeak on android, in order to try to make it sound like the espeak glenn variant, I set it to the edward variant with the volume raised up to 110 or 120 or so I increase it quite a bit... pitch raised to 70% or so, and pitch variation 20 or 30%. Unfortunately I can't do anything about the head-size. Glenn has a lower head size... I have learned to live with the current limitations although they are a bit frustrating. The way we did stuff in the 1990s was so much easier.

**Comment by joshknnd1982** [comment]
Ok so how do I add stuff to a repository. The github docs are not written in a step by step guide like the keysoft docs were back in 1990.

**Comment by joshknnd1982** [comment]
As a result of the new espeak variants, espeak ng is my default tts engine now. the new variants make espeak sound a whole lot better. I wish making a pull request on github were easier. Can someone on here make a pull request with the variants that I want included? Or maybe chat gpt can make a pull request for me somehow? I don't understand how pull requests work so as a result I doubt the new variants will ever be inclluded in espeak ng. ever.

**Comment by joshknnd1982** [comment]
chat gpt told me this. 1. Go to the website called GitHub.
2. Log in to your account.
3. Find the project you want to contribute to.
4. Click on the 'Fork' button near the top of the page.
5. This will create a copy of the project in your account.
6. Make changes to your copy of the project.
7. When you are done, click on the 'Compare & pull request' button.
8. Write a short message about the changes you made.
9. Click the 'Create pull request' button.
10. You have now made a pull request!

**Comment by joshknnd1982** [comment]
so based on that info, does github work like google drive you go into a folder and upload what you want into that folder?

**Comment by joshknnd1982** [comment]
ok this is a lot more complicated than google drive is. can someone just give me step by step directions so I can make a pull request please so the new variants can be added?

**Comment by beqabeqa473** [comment]
It is becoming a bit spammy. Please stop doing that as lots of emails are beeing sent by github and topic is nearly useless

**Comment by joshknnd1982** [comment]
I will stop being spammy when someone on here remotes into my machine with NVDA remote or jaws tandom and takes me step by step on how to make this pull request correctly so I can get the new variants into espeak ng. anyone want to volunteer?

**Comment by hozosch** [comment]
I already made a pull request now, but there are some checks failing, and I don't know what to do here. I guess there's only a limited number of voices allowed? I'm sorry if this pull request is too faulty.

**Comment by joshknnd1982** [comment]
thanks for making the pull request. I really appreciate it. in regards to how many variants... you can only have 230 or less, otherwise espeak becomes unstable and will crash.

**Comment by hozosch** [comment]
Well, with the new variants, we're at 239, and NVDA doesn't crash, but some languages disappear, more specifically the languages from the uralic branch, so Estonian, Finnish, Hungarian and lule saami. This is fixed when the overflowing variants are deleted.

**Comment by joshknnd1982** [comment]
that is why I made a zip file containing what I consider only the best variants, the total number of variants in that zip file is just 186.

**Comment by hozosch** [comment]
Ah, now I get it. However, for example, my iven4 variant is what I prefer to use, it would be gone when including it like this. It's a matter of taste. By that , I want to say that those variants may sound better than anything else to you, but not to anyone. I hope we can find a solution to satisfy anyone.

**Comment by joshknnd1982** [comment]
Maybe the overall code could be modified to allow for more variants to be included while allowing espeak to remain stable and functioning normally? Maybe we could increase the number of variants allowed to 500 or something like that?

**Comment by hozosch** [comment]
Yes, I think that's what's beeing looked at now over on my pull request.

**Comment by joshknnd1982** [comment]
I wish pull requests were more straightforward to make. then I could have made the pull request. I will ask freedom scientific if they could do a webinar on using jaws with github. That may help.
